spec/std/time/location_spec.cr

```diff
@@ -0,0 +1,265 @@
+        end
+      end
+
+      it "invalid timezone identifier" do
+        expect_raises(InvalidLocationNameError, "Foobar/Baz") do
+          Location.load("Foobar/Baz")
+        end
+      end
+
+      it "treats UTC as special case" do
+        Location.load("UTC").should eq Location::UTC
+        Location.load("").should eq Location::UTC
+
+        # Etc/UTC could be pointing to anything
+        Location.load("Etc/UTC").should_not eq Location::UTC
+      end
+
+      describe "validating name" do
+        it "absolute path" do
+          expect_raises(InvalidLocationNameError) do
+            Location.load("/America/New_York")
+          end
+          expect_raises(InvalidLocationNameError) do
+            Location.load("\\Zulu")
+          end
+        end
+        it "dot dot" do
+          expect_raises(InvalidLocationNameError) do
+            Location.load("../zoneinfo/America/New_York")
+          end
+          expect_raises(InvalidLocationNameError) do
+            Location.load("a..")
+          end
+        end
+      end
+
+      context "with ZONEINFO" do
+        it "loads from custom source" do
+          with_zoneinfo(File.join(__DIR__, "..", "data", "zoneinfo")) do
+            location = Location.load("Foo/Bar")
+            location.name.should eq "Foo/Bar"
+          end
+
+          with_zoneinfo(ZONEINFO_ZIP) do
+            location = Location.load("Asia/Jerusalem")
+            location.not_nil!.name.should eq "Asia/Jerusalem"
+          end
+        end
+
+        it "does not fall back to default sources" do
+          with_zoneinfo(File.join(__DIR__, "..", "data", "zoneinfo")) do
+            expect_raises(InvalidLocationNameError) do
+              location = Location.load("Europe/Berlin")
+            end
+          end
+
+          with_zoneinfo("nonexising_zipfile.zip") do
+            expect_raises(InvalidLocationNameError) do
+              location = Location.load("Europe/Berlin")
+            end
+          end
+        end
+      end
+    end
+
+    it "UTC" do
+      location = Location::UTC
+      location.name.should eq "UTC"
+
+      location.unspecified?.should be_false
+      location.utc?.should be_true
+      location.fixed?.should be_true
+
+      # this could fail if no source for localtime is available
+      unless Location.local.utc?
+        location.local?.should be_false
+      end
+
+      zone = location.lookup(Time.now)
+      zone.name.should eq "UTC"
+      zone.offset.should eq 0
+      zone.dst?.should be_false
+    end
+
+    it "UNSPECIFIED" do
+      location = Location::UNSPECIFIED
+      location.name.should eq "UNSPECIFIED"
+
+      location.unspecified?.should be_true
+      location.utc?.should be_false
+      location.fixed?.should be_false
+      location.local?.should be_false
+
+      zone = location.lookup(Time.now)
+      zone.name.should eq "UNSPECIFIED"
+      zone.offset.should eq 0
+      zone.dst?.should be_false
+    end
+
+    it ".local" do
```

Comment:
I concur it is confusing, but so would it be if `TC=""` *did not* mean `UTC` in contrast to `tzset`... I'm not sure what would be better, but in doubt I'd rather mirror the behaviour of `tzset` instead of introducing new semantics.

Reply:
I think the semantics of `TZ` environment variable should stay the same, we could perhaps remove `Location.load("")` to mean UTC.